### PR TITLE
Laboratory work #1, Andrew Zapekin - 23FPL1.

### PR DESCRIPTION
**Pull request by Andrew-cmd216**

### CODE REVIEW

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -1,11 +1,11 @@
 """
 Lab 1.
 
-Language detection
+Language detect
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:4]: revert

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
 """
 # pylint:disable=too-many-locals, unused-argument, unused-variable
 
-
+text = "Hey, how are you?"
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:8]: revert

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -20,6 +20,21 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if not text or not isinstance(text, str):
+        return None
+
+    text = text.lower()
+    num = '1234567890'
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:27]: can you combine num and symb?

File: lab_1_classify_by_unigrams/main.py [hunk 4]
```diff
@@ -20,6 +20,21 @@ def tokenize(text: str) -> list[str] | None:
...
+    symb = ',./?!:;#@*-&<>%'
+    tokens = []
+
+    for i in text:
+        if i not in num and i not in symb and i != ' ':
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:32]: can you replace i != ' ' somehow?

File: lab_1_classify_by_unigrams/main.py [hunk 5]
```diff
@@ -34,6 +49,33 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
 
     In case of corrupt input arguments, None is returned
     """
+    if not tokens or None:
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:52]: what are you checking here?

File: lab_1_classify_by_unigrams/main.py [hunk 6]
```diff
@@ -34,6 +49,33 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
...
+        return None
+
+    if not isinstance(tokens, list):
+        return None
+
+    token_freq = {}
+
+    for token in tokens:
+        if not isinstance(token, str):
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:61]: can you check it earlier?

File: lab_1_classify_by_unigrams/main.py [hunk 7]
```diff
@@ -49,7 +91,24 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
 
     In case of corrupt input arguments, None is returned
     """
+    if not language or not text or not isinstance(language, str):
+        return None
 
+    tokens = tokenize(text)
+
+    token_freq = calculate_frequencies(tokens)
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:99]: what if calculate_frequencies will return None?

File: lab_1_classify_by_unigrams/main.py [hunk 8]
```diff
@@ -49,7 +91,24 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
...
+
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:100]: delete whitespaces

File: lab_1_classify_by_unigrams/main.py [hunk 9]
```diff
@@ -49,7 +91,24 @@ def create_language_profile(language: str, text: str) -> dict[str, str | dict[st
...
+
+    return {
+        "name": language,
+        "freq": token_freq
+    }
+
+
+#language_name = 'en'
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:108]: delete useless comments

File: lab_1_classify_by_unigrams/main.py [hunk 10]
```diff
@@ -86,6 +156,8 @@ def compare_profiles(
     """
 
 
+
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:159]: ?

File: seminars/practice_2_strings.py [hunk 11]
```diff
@@ -15,24 +15,24 @@
 # strings are case-sensitive
 
 # Create a string
```
Comment (EIzharikov) [inline seminars/practice_2_strings.py:17]: revert chagnges in seminars

File: lab_1_classify_by_unigrams/start.py [hunk 12]
```diff
@@ -2,14 +2,15 @@
...
-
+import main as m
 
 def main() -> None:
     """
     Launches an implementation
     """
     with open("assets/texts/en.txt", "r", encoding="utf-8") as file_to_read_en:
         en_text = file_to_read_en.read()
+        m.create_language_profile('en', en_text)
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/start.py:13]: can you show the full work of your code here?

**Review by EIzharikov — COMMENTED** [review]
Nice objective! Make CI green and be ready for the next review! Good luck!

File: lab_1_classify_by_unigrams/main.py [hunk 13]
```diff
@@ -5,7 +5,6 @@
 """
 # pylint:disable=too-many-locals, unused-argument, unused-variable
 
-
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:8]: revert this change

File: lab_1_classify_by_unigrams/main.py [hunk 14]
```diff
@@ -20,6 +19,19 @@ def tokenize(text: str) -> list[str] | None:
...
+    if not text or not isinstance(text, str):
+        return None
+
+    text = text.lower()
+    symb = ',./?!:;#@*-&<>%1234567890 '
+    tokens = []
+
+    for i in text:
+        if i not in symb:
+            tokens.append(i)
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:31]: do you know what `isaplha()` is?

File: lab_1_classify_by_unigrams/main.py [hunk 15]
```diff
@@ -107,6 +168,7 @@ def detect_language(
     """
 
 
+
```
Comment (EIzharikov) [inline lab_1_classify_by_unigrams/main.py:171]: delete this whitespace

File: seminars/practice_1_run_debug.py [hunk 16]
```diff
@@ -1,42 +0,0 @@
-"""
-Programming 2024
-Seminar 1
-Running python application and debugging
```
Comment (EIzharikov) [inline seminars/practice_1_run_debug.py:4]: revert all changes from seminars

File: seminars/practice_2_strings.py [hunk 17]
```diff
@@ -1,200 +0,0 @@
-"""
-Programming 2024
-Seminar 2
-
-
-Data Type: String
-"""
```
Comment (EIzharikov) [inline seminars/practice_2_strings.py:7]: the same

File: seminars/practice_3_lists.py [hunk 18]
```diff
@@ -1,193 +0,0 @@
-"""
-Programming 2024
-Seminar 3
```
Comment (EIzharikov) [inline seminars/practice_3_lists.py:3]: .

**Review by EIzharikov — COMMENTED** [review]
Work is good, bou you should hurry up! FIx and answer to ALL my comments, make CI green and be ready for the lab work defense!

File: lab_1_classify_by_unigrams/main.py (hunk 1 above)
Comment (Andrew-cmd216) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 2 above)
Comment (Andrew-cmd216) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 3 above)
Comment (Andrew-cmd216) [reply lab_1_classify_by_unigrams/main.py]: did it

File: lab_1_classify_by_unigrams/main.py (hunk 4 above)
Comment (Andrew-cmd216) [reply lab_1_classify_by_unigrams/main.py]: fixed in symb

File: lab_1_classify_by_unigrams/main.py (hunk 5 above)
Comment (Andrew-cmd216) [reply lab_1_classify_by_unigrams/main.py]: reverted it cuz it's useless

File: lab_1_classify_by_unigrams/main.py (hunk 8 above)
Comment (Andrew-cmd216) [reply lab_1_classify_by_unigrams/main.py:94]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 9 above)
Comment (Andrew-cmd216) [reply lab_1_classify_by_unigrams/main.py]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 10 above)
Comment (Andrew-cmd216) [reply lab_1_classify_by_unigrams/main.py:148]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 15 above)
Comment (Andrew-cmd216) [reply lab_1_classify_by_unigrams/main.py:171]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 14 above)
Comment (Andrew-cmd216) [reply lab_1_classify_by_unigrams/main.py:31]: fixed

File: seminars/practice_3_lists.py (hunk 18 above)
Comment (Andrew-cmd216) [reply seminars/practice_3_lists.py]: fixed

File: seminars/practice_2_strings.py (hunk 17 above)
Comment (Andrew-cmd216) [reply seminars/practice_2_strings.py]: fixed

File: seminars/practice_1_run_debug.py (hunk 16 above)
Comment (Andrew-cmd216) [reply seminars/practice_1_run_debug.py]: fixed

File: seminars/practice_2_strings.py (hunk 11 above)
Comment (Andrew-cmd216) [reply seminars/practice_2_strings.py]: fixed

File: lab_1_classify_by_unigrams/start.py (hunk 12 above)
Comment (Andrew-cmd216) [reply lab_1_classify_by_unigrams/start.py]: fixed
